### PR TITLE
feat(java): detect & rename members named after reserved words

### PR DESCRIPTION
**Pull request by bmaizels**
---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: packages/jsii-pacmak/lib/targets/java.ts [hunk 1]
```diff
@@ -183,6 +184,50 @@ interface JavaProp {
...
+
+        if (JavaGenerator.RESERVED_KEYWORDS.includes(propertyName)) {
+            return `${propertyName}Value`;
+        } else {
+            return propertyName;
+        }
+    }
+
+    /**
+     * Turns a raw javascript method name (eg: 'import') into a safe Java method name 9eg: 'doImport').
```
Comment (RomainMuller) [inline packages/jsii-pacmak/lib/targets/java.ts]: 9eg?

File: packages/jsii-pacmak/lib/targets/java.ts [hunk 2]
```diff
@@ -183,6 +184,50 @@ interface JavaProp {
...
+     * Turns a raw javascript property name (eg: 'default') into a safe Java property name (eg: 'defaultValue').
+     * @param propertyName the raw JSII property Name
+     */
+    private static safeJavaPropertyName(propertyName: string) {
+        if (!propertyName) {
+            return propertyName;
+        }
+
+        if (JavaGenerator.RESERVED_KEYWORDS.includes(propertyName)) {
+            return `${propertyName}Value`;
```
Comment (RomainMuller) [inline packages/jsii-pacmak/lib/targets/java.ts]: Might be worth checking whether this is the name of an existing property on the same type... If that is so, just bail out saying you're out of options there...

File: packages/jsii-pacmak/lib/targets/java.ts [hunk 3]
```diff
@@ -183,6 +184,50 @@ interface JavaProp {
...
+     * Turns a raw javascript method name (eg: 'import') into a safe Java method name 9eg: 'doImport').
+     * @param methodName
+     */
+    private static safeJavaMethodName(methodName: string) {
+        if (!methodName) {
+            return methodName;
+        }
+
+        if (JavaGenerator.RESERVED_KEYWORDS.includes(methodName)) {
+            return `do${toPascalCase(methodName)}`;
```
Comment (RomainMuller) [inline packages/jsii-pacmak/lib/targets/java.ts]: Same as above, would be nice to have a collision check.

File: packages/jsii-pacmak/lib/targets/java.ts (hunk 1 above)
Comment (bmaizels) [reply packages/jsii-pacmak/lib/targets/java.ts]: About to be fixed.

File: packages/jsii-pacmak/lib/targets/java.ts (hunk 2 above)
Comment (bmaizels) [reply packages/jsii-pacmak/lib/targets/java.ts]: Difficult to do, so in the interests of unblocking CDK builds moving ahead without this functionality for now.

File: packages/jsii-pacmak/lib/targets/java.ts (hunk 3 above)
Comment (bmaizels) [reply packages/jsii-pacmak/lib/targets/java.ts]: Difficult to do, so in the interests of unblocking CDK builds moving ahead without this functionality for now.

**Comment by RomainMuller** [comment]
Looks good - just waiting for a confirmation build of the CDK to succeed before I approve this... To be safe...

**Pull request merged**